pkg/awsvpc/nacl_analysis.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package awsvpc
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/ipblock"
+
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/commonvpc"
+)
+
+type AWSNACLAnalyzer struct {
+	naclResource       *types.NetworkAcl
+	referencedIPblocks []*ipblock.IPBlock
+}
+
+func NewAWSNACLAnalyzer(nacl *types.NetworkAcl) *AWSNACLAnalyzer {
+	res := &AWSNACLAnalyzer{naclResource: nacl}
+	return res
+}
+
+func (na *AWSNACLAnalyzer) GetNumberOfRules() int {
+	return len(na.naclResource.Entries)
+}
+
+func (na *AWSNACLAnalyzer) Name() *string {
+	return na.naclResource.NetworkAclId
+}
+
+func (na *AWSNACLAnalyzer) ReferencedIPblocks() []*ipblock.IPBlock {
+	return na.referencedIPblocks
+}
+
+func (na *AWSNACLAnalyzer) GetNACLRule(index int) (ruleStr string, ruleRes *commonvpc.NACLRule, isIngress bool, err error) {
+	var conns *connection.Set
+	var connStr string
+	ruleObj := na.naclResource.Entries[index]
+	*ruleObj.Protocol = convertProtocol(*ruleObj.Protocol)
```

Comment:
```suggestion
	protocol := convertProtocol(*ruleObj.Protocol)
```
Better leave the AWS-API object as-is.